### PR TITLE
Ignore additional properties

### PR DESCRIPTION
**Pull request by bjgill**
### PR checklist

- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Ran the shell script under `./bin/` to update Petstore sample so that CIs can verify the change. (For instance, only need to run `./bin/{LANG}-petstore.sh` and `./bin/security/{LANG}-petstore.sh` if updating the {LANG} (e.g. php, ruby, python, etc) code generator or {LANG} client's mustache templates). Windows batch files can be found in `.\bin\windows\`.
- [x] Filed the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`, `4.0.x`. Default: `master`.
- [x] Copied the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language. @frol @farcaller 

### Description of the PR

This is a workaround for https://github.com/OpenAPITools/openapi-generator/issues/318. In swagger-codegen, we did not support `additionalProperties` and so ignored them. 

In openapi-generator, we now get back a type (`map`), where we got back a null previously. `postProcessModels` doesn't have enough context to work out what sort of map is required (we don't know the types of the `additionalProperties`).

Thus, this workaround detects this specific failure case and fiddles with the `dataType` to get `additionalProperties` ignored again. This will allow us to generate from specs containing `additionalProperties` and have the resultant crates compile.

I've added in a warning to users to let them know what's happening.

I'm building on top of https://github.com/OpenAPITools/openapi-generator/pull/658 to allow us to add a test to ensure that we don't regress this in future.

### CODE REVIEW

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java [hunk 1]
```diff
@@ -1030,21 +1030,28 @@ static String matchingIntType(boolean unsigned, Long inclusiveMin, Long inclusiv
...
                 // Object isn't a sensible default. Instead, we set it to
                 // 'null'. This ensures that we treat this model as a struct
                 // with multiple parameters.
                 cm.dataType = null;
             } else if (cm.dataType != null) {
-                // We need to hack about with single-parameter models to get
-                // them recognised correctly.
-                cm.isAlias = false;
-                cm.dataType = typeMapping.get(cm.dataType);
+                if (cm.dataType == "map") {
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java]: @bjgill should this be `"map".equals(cm.dataType)` instead?

Comment (bjgill) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/RustServerCodegen.java]: Good point - I'll be consistent with the equivalent 5 lines previously, though - `cm.dataType.equals("map")`

**Comment by bjgill** [comment]
Not sure why travis failed. Doesn't seem to be anything Rust-related, though. Do you have the ability to retry the job?

```
...
npm WARN deprecated jade@0.26.3: Jade has been renamed to pug, please install the latest version of pug instead of jade
npm WARN deprecated formatio@1.1.1: This package is unmaintained. Use @sinonjs/formatio instead
npm WARN deprecated minimatch@0.2.14: Please update to minimatch 3.0.2 or higher to avoid a RegExp DoS issue
npm WARN deprecated graceful-fs@2.0.3: please upgrade to graceful-fs 4 for compatibility with current and future versions of Node.js
npm WARN deprecated babel@6.23.0: In 6.x, the babel package has been deprecated in favor of babel-cli. Check https://opencollective.com/babel to support the Babel maintainers
No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.
Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#Build-times-out-because-no-output-was-received
The build has been terminated
```

**Comment by wing328** [comment]
@bjgill Restarted the job. Let's see how it goes

**Comment by bjgill** [comment]
That passed, thanks.

**Pull request merged**